### PR TITLE
Add segmented buttons

### PR DESCRIPTION
**Pull request by smialko**
I've created a segmented picker button for Calendar View. Right now, it's purely the UI, ready and waiting for the logic to be implemented in the future. I've taken into account the potential variation in the number of buttons it might contain, so when you test it, that's something to bear in min

screenshot_2024-01-02_at_11 33 41

### CODE REVIEW

File: app/javascript/react/components/SegmentedPickerButtons/SegmentedPickerButtons.style.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  border-radius: 5px;
+`;
+
+interface SegmentedOptionProps {
+  isSelected: boolean;
+  isFirst: boolean;
+  isLast: boolean;
+}
+
+const SegmentedOption = styled.div<SegmentedOptionProps>`
```
Comment (jankrzempek) [inline app/javascript/react/components/SegmentedPickerButtons/SegmentedPickerButtons.style.tsx:18]: The given approach shows that our `SegmentedPicker` does need to know what means `first` and `last`. There is no need for this, as only style needs this. I would personally pass high-level info and decompose it later. I would like to suggest something below. One more thing is the conditional `border-radius` a bit too complicated I think 😅

NOI => Number of items

```
interface SegmentedOptionProps {
  isSelected: boolean;
  index: number;
  NOI: number;
}

interface Border {
  index: number;
  NOI: number;
}

const getBorder = ({ index, NOI }: Border) => {
  const leftBorder = "5px 0 0 5px"
  const rightBorder = "0px 5px 5px 0px"

  if (index == NOI - 1) {
    return rightBorder
  } else if (index == 0) {
    return leftBorder
  }
}

const SegmentedOption = styled.div<SegmentedOptionProps>`
  padding: 8px 15px;
  margin: 2px;
  cursor: pointer;
  background-color: ${({ isSelected }) => (isSelected ? blue : "transparent")};
  border-radius: ${({ isSelected, index, NOI }) => isSelected && getBorder({index, NOI})};
  color: ${({ isSelected }) => (isSelected ? blue : "#fff")};
  > span {
    color: ${({ isSelected }) => (isSelected ? "#fff" : lightGray)};
    text-transform: uppercase;
  }
`;




<S.SegmentedOption
          key={option}
          onClick={() => onOptionChange(option)}
          isSelected={option === selectedOption}
          index={index}
          NOI={options.length}
        >
          <span>{option}</span>
        </S.SegmentedOption>
```

**Review by jankrzempek — COMMENTED** [review]
Works well, for longer and shorter strings. Nice job done in here! 👮‍♂️

File: app/javascript/react/components/SegmentedPickerButtons/SegmentedPickerButtons.style.tsx (hunk 1 above)
Comment (smialko) [reply app/javascript/react/components/SegmentedPickerButtons/SegmentedPickerButtons.style.tsx:18]: Good point, thanks <3

File: app/javascript/react/components/SegmentedPickerButtons/SegmentedPickerButtons.style.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    isSelected
+      ? isFirst
+        ? "5px 0 0 5px"
+        : isLast
+        ? "0 5px 5px 0"
+        : "0px"
+      : "0px"};
+  color: ${({ isSelected }) => (isSelected ? blue : "#fff")};
+  > span {
+    color: ${({ isSelected }) => (isSelected ? "#fff" : lightGray)};
```
[the remark this answers is not in the document]
Comment (smialko) [reply app/javascript/react/components/SegmentedPickerButtons/SegmentedPickerButtons.style.tsx:33]: Sure, you're right 🫠

**Review by jankrzempek — APPROVED** [review]
Nothing that I am so sure about, just suggesting some more possibilities. Great picker, great job! 🙌🏻

File: app/javascript/react/pages/CalendarPage.tsx [hunk 3]
```diff
@@ -24,6 +25,7 @@ const CalendarPage = () => {
       <Navbar />
       <PageLayout>
         <StationDataContainer>
+          <SegmentedPicker options={["Day", "Week", "Month"]} />
```
[the remark this answers is not in the document]
Comment (smialko) [reply app/javascript/react/pages/CalendarPage.tsx:28]: Ugh, they're not supposed to be there yet, and they're not even real - my mocks for testing 🙈 Good catch 😅

**Pull request merged**